modules/host/network.go

```diff
@@ -49,11 +49,11 @@ func (h *Host) initNetworking(address string) (err error) {
 		return err
 	}
 	// Automatically close the listener when h.tg.Stop() is called.
+	err = h.tg.Add()
+	if err != nil {
+		return err
+	}
 	go func() {
-		err := h.tg.Add()
-		if err != nil {
-			return
-		}
```

Comment:
OH, it's blocking on `listener.Accept` because the listener closer thread was not created in time. You're right, that's a deadlock.